core/commands/pin/remotepin.go

```diff
@@ -0,0 +1,614 @@
+	Helptext: cmds.HelpText{
+		Tagline: "List objects pinned to remote pinning service.",
+		ShortDescription: `
+Returns a list of objects that are pinned to a remote pinning service.
+`,
+		LongDescription: `
+Returns a list of objects that are pinned to a remote pinning service.
+`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("request-id", false, true, "Request ID of the pin to be listed."),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption(pinNameOptionName, "Return pins objects with names that contain provided value (case-sensitive, exact match)."),
+		cmds.StringsOption(pinCIDsOptionName, "Return only pin objects for the specified CID(s); optional, comma separated."),
+		cmds.StringsOption(pinStatusOptionName, "Return only pin objects with the specified statuses; optional, comma separated."),
+		cmds.StringOption(pinServiceNameOptionName, "Name of the remote pinning service to use."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx, cancel := context.WithCancel(req.Context)
+		defer cancel()
+
+		service, _ := req.Options[pinServiceNameOptionName].(string)
+		c, err := getRemotePinServiceOrEnv(env, service)
+		if err != nil {
+			return err
+		}
+
+		psCh, errCh, err := lsRemote(ctx, req, c)
+		if err != nil {
+			return err
+		}
+
+		for ps := range psCh {
+			if err := res.Emit(toRemotePinOutput(ps)); err != nil {
+				return err
+			}
+		}
+
+		return <-errCh
+	},
+	Type: RemotePinOutput{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *RemotePinOutput) error {
+			// pin remote ls produces a flat output similar to legacy pin ls
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", out.RequestID, out.Cid, out.Status, out.Name)
+			return nil
+		}),
+	},
+}
+
+// Returns pin objects with passed RequestIDs or matching filters.
+func lsRemote(ctx context.Context, req *cmds.Request, c *pinclient.Client) (chan pinclient.PinStatusGetter, chan error, error) {
+	// If request-ids are provided, do direct lookup for specific objects
+	if len(req.Arguments) > 0 {
+		return lsRemoteByRequestId(ctx, req, c)
+	}
+	// else, apply filters to find matching pin objects
+	return lsRemoteWithFilters(ctx, req, c)
+}
+
+// Executes GET /pins/{requestid} for each requestid passed as argument.
+// Status checks are executed one by one and operation is aborted on first error.
+func lsRemoteByRequestId(ctx context.Context, req *cmds.Request, c *pinclient.Client) (chan pinclient.PinStatusGetter, chan error, error) {
+	lsIDs := req.Arguments
+	res := make(chan pinclient.PinStatusGetter, 1)
+	errs := make(chan error, 1)
+	go func() {
+		defer close(res)
+		defer close(errs)
+		for _, requestId := range lsIDs {
+			ps, err := c.GetStatusByID(ctx, requestId)
+			if err != nil {
+				errs <- err
+				return
+			}
+			select {
+			case res <- ps:
+			case <-ctx.Done():
+				errs <- ctx.Err()
+				return
+			}
+		}
+	}()
+	return res, errs, nil
+}
```

Comment:
@petar @aschmahmann  would appreciate quick review of `lsRemoteByRequestId` – I don't feel fluent in golang yet.

Reply:
This function seems fine, although from here it looks like we should really just remove GetStatusByID and bundle it up with normal pin listing. WDYT?